ext/dom/element.c

```diff
@@ -842,6 +842,44 @@ PHP_METHOD(Dom_Element, getElementsByTagName)
 }
 /* }}} end dom_element_get_elements_by_tag_name */
 
+PHP_METHOD(Dom_Element, getElementsByClassName)
+{
+	dom_object *intern, *namednode;
+	zend_string *class_names;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "P", &class_names) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	if (ZSTR_LEN(class_names) > INT_MAX) {
+		zend_argument_value_error(1, "is too long");
```

Comment:
Probably an error message we should improve, but this feels this is pre-existing

Reply:
Probably. Note that we have the "is too long" wording in other extensions too. So if we take care of this, we should change it consistently. Or maybe even introduce a common Zend API to make the wording consistent, like you did with the "must not be empty" message?
Anyway, besides the error message, I believe this condition is not really going to be hit in practice. Is anyone going to make a class name list that spans 2**31 characters? :thinking: